test/whl-scale-tests/common.ps1

```diff
@@ -0,0 +1,66 @@
+$ErrorActionPreference = 'Stop'
+
+function Get-RandomPassword {
+    param (
+        [Parameter(Mandatory)]
+        [ValidateRange(4,[int]::MaxValue)]
+        [int] $length,
+        [int] $upper = 1,
+        [int] $lower = 1,
+        [int] $numeric = 1,
+        [int] $special = 1
+    )
+    if($upper + $lower + $numeric + $special -gt $length) {
+        throw "number of upper/lower/numeric/special char must be lower or equal to length"
+    }
+    $uCharSet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+    $lCharSet = "abcdefghijklmnopqrstuvwxyz"
+    $nCharSet = "0123456789"
+    $sCharSet = "{-)}^%(_!#$"
+    $charSet = ""
+    if($upper -gt 0) { $charSet += $uCharSet }
+    if($lower -gt 0) { $charSet += $lCharSet }
+    if($numeric -gt 0) { $charSet += $nCharSet }
+    if($special -gt 0) { $charSet += $sCharSet }
+    
+    $charSet = $charSet.ToCharArray()
+    $rng = New-Object System.Security.Cryptography.RNGCryptoServiceProvider
+    $bytes = New-Object byte[]($length)
+    $rng.GetBytes($bytes)
+ 
+    $result = New-Object char[]($length)
+    for ($i = 0 ; $i -lt $length ; $i++) {
+        $result[$i] = $charSet[$bytes[$i] % $charSet.Length]
+    }
+    $password = (-join $result)
+    $valid = $true
+    if($upper   -gt ($password.ToCharArray() | Where-Object {$_ -cin $uCharSet.ToCharArray() }).Count) { $valid = $false }
+    if($lower   -gt ($password.ToCharArray() | Where-Object {$_ -cin $lCharSet.ToCharArray() }).Count) { $valid = $false }
+    if($numeric -gt ($password.ToCharArray() | Where-Object {$_ -cin $nCharSet.ToCharArray() }).Count) { $valid = $false }
+    if($special -gt ($password.ToCharArray() | Where-Object {$_ -cin $sCharSet.ToCharArray() }).Count) { $valid = $false }
+ 
+    if(!$valid) {
+         $password = Get-RandomPassword $length $upper $lower $numeric $special
+    }
+    return $password
+}
+
+
+function SubstituteNameValuePairs {
+    param(
+        [string][Parameter(Mandatory=$true)] $InputFilePath,
+        [string][Parameter(Mandatory=$true)] $OutputFilePath,
+        [hashtable][Parameter(Mandatory=$true)] $Substitutions
+    )
+
+    # Ensure the input file exists
+    if (-not (Test-Path -Path $InputFilePath)) {
+        Write-Host "  Input File: '$InputFilePath' does not exist" -ForegroundColor Red
+        exit
+    }
+
+    foreach($subItem in $Substitutions.GetEnumerator())
+    {
+        (Get-Content $InputFilePath).Replace($subItem.Name, $subItem.Value) | Set-Content $OutputFilePath

```

Comment:
This only works when input file and output file are the same since each replace is being applied to the original input file. 

Maybe something like this instead, which also reduces number of file reads/writes
```
$content = Get-Content $InputFilePath;
foreach($subItem in $Substitutions.GetEnumerator())
{
    $content = $content.Replace($subItem.Name, $subItem.Value); 
}
$content | Set-Content $OutputFilePath;
```

Reply:
Good catch

Reply:
Fixed

Reply:
Fixed